node/impl/full/multisig.go

```diff
@@ -0,0 +1,282 @@
+package full
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/actors"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	init_ "github.com/filecoin-project/specs-actors/actors/builtin/init"
+	samsig "github.com/filecoin-project/specs-actors/actors/builtin/multisig"
+
+	"github.com/ipfs/go-cid"
+	"github.com/minio/blake2b-simd"
+	cbg "github.com/whyrusleeping/cbor-gen"
+
+	"go.uber.org/fx"
+	"golang.org/x/xerrors"
+)
+
+type MsigAPI struct {
+	fx.In
+
+	WalletAPI WalletAPI
+	StateAPI  StateAPI
+	MpoolAPI  MpoolAPI
+}
+
+func (a *MsigAPI) MsigCreate(ctx context.Context, req int64, addrs []address.Address, val types.BigInt, src address.Address, gp types.BigInt) (api.AddressPair, error) {
+
+	lenAddrs := int64(len(addrs))
+
+	if lenAddrs < req {
+		return api.UndefAddressPair, xerrors.Errorf("cannot require signing of more addresses than provided for multisig")
+	}
+
+	if req == 0 {
+		req = lenAddrs
+	}
+
+	if src == address.Undef {
```

Comment:
I would do this logic on the client side, and require callers of the API to always pass an address here.